metadata-ingestion/src/datahub/ingestion/source/hex/mapper.py

```diff
@@ -0,0 +1,332 @@
+                    )
+                    for cat in categories
+                ]
+            )
+        if collections and self._collections_as_tags:
+            tag_associations.extend(
+                [
+                    TagAssociationClass(
+                        tag=make_tag_urn(tag=f"hex:collection:{col.name}")
+                    )
+                    for col in collections
+                ]
+            )
+
+        return GlobalTagsClass(tags=tag_associations) if tag_associations else None
+
+    def _ownership(
+        self, creator: Optional[Owner], owner: Optional[Owner]
+    ) -> Optional[OwnershipClass]:
+        if self._set_ownership_from_email:
+            # since we are not making any diff of creator/owner, we usually have duplicates
+            # TODO: set or ownership types to properly differentiate them, maybe by config?
+            unique_owners = set(o for o in [creator, owner] if o)
+            owners: List[OwnerClass] = [
+                OwnerClass(owner=make_user_urn(o.email), type=DEFAULT_OWNERSHIP_TYPE)
+                for o in unique_owners
+            ]
+
+            return OwnershipClass(owners=owners) if owners else None
+        return None
+
+    def _dashboard_usage_statistics(
+        self, analytics: Optional[Analytics]
+    ) -> Optional[DashboardUsageStatisticsClass]:
+        # we are mapping here only the last 7 days of views
+        # we could emit multiple events for different time windows or make which metric to track configurable
+        if analytics and (analytics.appviews_last_7_days or analytics.last_viewed_at):
+            return DashboardUsageStatisticsClass(
+                timestampMillis=make_ts_millis(datetime.now()),
+                viewsCount=analytics.appviews_last_7_days,
+                eventGranularity=TimeWindowSizeClass(
+                    unit=CalendarIntervalClass.WEEK, multiple=1
+                ),
+                lastViewedAt=make_ts_millis(analytics.last_viewed_at),
+            )
+        return None
+
+    def _platform_instance_aspect(self) -> DataPlatformInstanceClass:
+        return DataPlatformInstanceClass(
+            platform=make_data_platform_urn(HEX_PLATFORM_NAME),
+            instance=make_dataplatform_instance_urn(
+                platform=HEX_PLATFORM_NAME, instance=self._platform_instance
+            )
+            if self._platform_instance
+            else None,
+        )
+
+    def _yield_mcps(
+        self, entity_urn: Urn, aspects: List[_Aspect]
+    ) -> Iterable[MetadataWorkUnit]:
+        for mcpw in MetadataChangeProposalWrapper.construct_many(
+            entityUrn=entity_urn.urn(),
+            aspects=aspects,
+        ):
+            wu = MetadataWorkUnit.from_metadata(metadata=mcpw)
+            maybe_wu = self._maybe_patch_wu(wu)
+            if maybe_wu:
+                yield maybe_wu
+
+    def _maybe_patch_wu(self, wu: MetadataWorkUnit) -> Optional[MetadataWorkUnit]:
```

Comment:
now that we're planning on pulling the queries in as part of the hex ingestion directly, I don't think we _need_ the ability to patch DashboardInfo. It's still a nice to have though

Reply:
true, that's why I have set `patch_metadata: False` by default